add GetAddressBalance gRPC

LarryRuane opened the pull request:
Closes #294. This is part of taddr (transparent) support. Still needs unit test and darksidewalletd support, but does work with manual testing, just want to make sure this interface is okay (@gmale, @pacu) before writing tests. You can try it quickly (mainnet):
```
$ grpcurl -plaintext -d '{"addresses":["t3WDhPfik343yNmPTqtkZAoQZeqA83K7Y3f"]}' localhost:9067 cash.z.wallet.sdk.rpc.CompactTxStreamer/GetTaddressBalance
{
  "zatoshis": "2817000000000"
}
$ 
```
(You can specify multiple addresses.) This returns the balance (the same as returned by `getaddressbalance` zcashd rpc) for a single address. The streaming variation is useful if there are a large number of addresses; You can use standard input by specifying `-d @`, for example:
```
$ grpcurl -plaintext -d @ localhost:9067 cash.z.wallet.sdk.rpc.CompactTxStreamer/GetTaddressBalanceStream <<EOF
{"address":"t3WDhPfik343yNmPTqtkZAoQZeqA83K7Y3f"}
{"address":"t1bzsJ7mRyHwsuXEPMud32SFTjLJ7FrWfpf"}
EOF
{
  "zatoshis": "2959250000000"
}
$ 
```

pacu commented on walletrpc/service.proto:
```diff
@@ -86,6 +86,13 @@ message PingResponse {
+    string address = 1;
+}
+message Amount {
+    int64 zatoshis = 1;
```
shouldn't this be just "zatoshi"

gmale replied:
+1

gmale replied:
or even just "balance" in case we later do decide to add in "received" and want to do less refactoring at that time. I think `zatoshi` is fine though because it is more descriptive.

LarryRuane replied:
On the topic of the name of this field, we should probably follow the convention used in `zcashd` rpcs, which is `valueZat`.

gmale commented on walletrpc/service.proto:
```diff
@@ -101,6 +108,7 @@ service CompactTxStreamer {
 
     // Return the txids corresponding to the given t-address within the given block range
     rpc GetAddressTxids(TransparentAddressBlockFilter) returns (stream RawTransaction) {}
+    rpc GetAddressBalance(stream Address) returns (Amount) {}
```
Where did we land on discussing whether this should be `stream` or `repeated`? Were there any arguments for making this a stream? I think passing an array of addresses, rather than a stream, might be slightly easier on the client side because you can make it a blocking or non-blocking call.

I believe passing an array here would let clients create simple [unary rather than streaming requests](https://grpc.io/docs/what-is-grpc/core-concepts/#client-streaming-rpc).

gmale replied:
Doing so might involve using a wrapper message such as:

```
message AddressList {
    repeated Address addresses = 1;
}

. . .

 rpc GetAddressBalance(AddressList) returns (Amount) {}
```

which is probably fine (unless there is an easier way).

LarryRuane commented:
Addressed review comments. The `GetAddressBalance` gRPC argument is now `repeated` (array) instead of streaming. Here's a mainnet usage example:
```
$ grpcurl -plaintext -d '{"address":["t3WDhPfik343yNmPTqtkZAoQZeqA83K7Y3f"]}' localhost:9067 cash.z.wallet.sdk.rpc.CompactTxStreamer/GetAddressBalance
{
  "valueZat": "3614125000000"
}
$ 
```
The JSON list can have multiple entries, of course.

Also, as you can see above, the field name is now `valueZat` to be consistent with zcashd.

Just in case the streaming version of this gRPC turns out to be useful in the future, I preserved it as `GetAddressBalanceStream`. The examples given in the [description](https://github.com/zcash/lightwalletd/pull/295#issue-464128498) will work with this rename.

gmale commented on walletrpc/service.proto:
```diff
@@ -101,6 +111,8 @@ service CompactTxStreamer {
 
     // Return the txids corresponding to the given t-address within the given block range
     rpc GetAddressTxids(TransparentAddressBlockFilter) returns (stream RawTransaction) {}
+    rpc GetAddressBalance(AddressList) returns (Balance) {}
```
was this comment addressed? I think you made a good point here and it might be ideal to make it clear in code that this does not work for zaddrs. Perhaps we do the same as the `getAddresTxids` call and name the parameter in such a way as to make that distinction clear:
```
rpc GetAddressBalance(TransparentAddressList) returns (Balance) {}
rpc GetAddressBalanceStream(stream TransparentAddress) returns (Balance) {}
```
Personally, I think I prefer your suggestion of just calling these `GetTaddressBalance` and `GetTaddressBalanceStream`
